lib/rules/no-restricted-service-injections.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const kebabCase = require('lodash.kebabcase');
+const assert = require('assert');
+const emberUtils = require('../utils/ember');
+
+const DEFAULT_ERROR_MESSAGE = 'Injecting this service is not allowed from this file.';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow injecting certain services under certain paths',
+      category: 'Miscellaneous',
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-restricted-service-injections.md',
+    },
+    schema: {
+      type: 'array',
+      minItems: 1,
+      items: [
+        {
+          type: 'object',
+          required: ['paths', 'services'],
+          properties: {
+            paths: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+              },
+            },
+            services: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+              },
+            },
+            error: {
+              type: 'string',
+            },
+          },
+          additionalProperties: false,
+        },
+      ],
+    },
+  },
+
+  DEFAULT_ERROR_MESSAGE,
+
+  create(context) {
+    // Validate options.
+    context.options.forEach((option) =>
+      option.services.forEach((service) =>
+        assert(
+          service.toLowerCase() === service,
+          'Service name should be passed in kebab-case (all lower case)'
+        )
+      )
+    );
+
+    // Find matching blacklist entries for this file path.
+    const blacklists = context.options.filter((option) =>
+      option.paths.some((path) => context.getFilename().match(path))
+    );
+
+    if (!blacklists || blacklists.length === 0) {
```

Comment:
You shouldn't need the `!blacklists` check.

Reply:
Fixed! Thanks.